packages/caravan-wallets/src/bitbox.ts

```diff
@@ -0,0 +1,551 @@
+/*** This module provides classes for BitBox hardware wallets.
+ *
+ * The following API classes are implemented:
+ *
+ * * BitBoxGetMetadata
+ * * BitBoxExportPublicKey
+ * * BitBoxExportExtendedPublicKey
+ * * BitBoxSignMultisigTransaction
+ */
+
+import {
+  BitcoinNetwork,
+  getPsbtVersionNumber,
+  PsbtV2,
+  ExtendedPublicKey,
+  MultisigAddressType,
+} from "@caravan/bitcoin";
+import {
+  ACTIVE,
+  PENDING,
+  INFO,
+  DirectKeystoreInteraction,
+} from "./interaction";
+
+import { MultisigWalletConfig } from "./types";
+
+import {
+  BtcCoin,
+  BtcMultisigScriptType,
+  BtcScriptConfig,
+  PairedBitBox,
+} from 'bitbox-api';
+
+/**
+ * Constant defining BitBox interactions.
+ */
+export const BITBOX = "bitbox";
+
+// Callback for showing the BitBox pairing code. It returns a function to hide it again.
+export type TShowPairingCode = (pairingCode: string) => (() => void) | null;
+
+function convertNetwork(network: BitcoinNetwork): BtcCoin {
+  return network === 'mainnet' ? 'btc' : 'tbtc';
+}
+
+function convertToBtcMultisigScriptType(addressType: MultisigAddressType): BtcMultisigScriptType {
+  switch (addressType) {
+    case 'P2WSH':
+      return 'p2wsh';
+    case 'P2SH-P2WSH':
+      return 'p2wshP2sh';
+    default:
+      throw new Error(`BitBox does not support ${addressType} multisig.`);
+  }
+}
+
+async function convertMultisig(pairedBitBox: PairedBitBox, walletConfig: MultisigWalletConfig): Promise<{ scriptConfig: BtcScriptConfig; keypathAccount: string; }> {
```

Comment:
It feels like this function is doing two things and it might cleaner to separate them. Correct me if I'm wrong but what it's doing is:

1. verifying that the devices root fingerprint is in a given wallet config
2. converting the caravan style wallet config to bitbox style 

What if the verification was a method on the interaction base class where you know you'll have a paired bitbox you can interact with and then this was kept as a simple conversion utility function? 

Having the verification on the interaction would allow it to be used in other interactions as well. could even retrieve the root fingerprint presumably on startup and store it as a property on the class or something and then use it as needed. Not sure if that's strictly correct or necessary, but I did find this a little hard to grok the way it is.

Reply:
> It feels like this function is doing two things and it might cleaner to separate them. Correct me if I'm wrong but what it's doing is:

>    1. verifying that the devices root fingerprint is in a given wallet config
>    2. converting the caravan style wallet config to bitbox style

It is not doing the first one. It is merely trying to identify which of the xpubs belongs to the device, as  that is a required parameter in the script config passed to the BitBox (`ourXpubIndex`).